### PR TITLE
Some miscellaneous repo cleanup

### PR DESCRIPTION
**Pull request by konklone**
This does a few things:

* Moves index markdown files out of `_includes/` and into either the `pages/` directory or their respective collections directory (e.g. `_userconfig`), so that the `_includes/` directory only includes template fragments.
* Deletes some unused files from the project root, to make it cleaner and easier for new people to understand the project and how to work in it.
* Versions `Gemfile.lock`, so that you have a much higher chance of other users using identical gem versions to the ones the owners are using.

My editor is configured to trim trailing whitespace automatically, so there are some whitespace deletions in the diff. You can add `?w=1` to the end of the GitHub URL for this pull request to suppress whitespace changes from the visual diff.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 _site/
-Gemfile.lock
-*~
\ No newline at end of file
+*~
+.DS_Store
```
Comment (konklone) [inline .gitignore]: I figured I would also ignore `.DS_Store` while I was at it.

**Comment by konklone** [comment]
In removing non-includes from `_includes/`, I was also able to remove some of the `index.html` templates that "included" the .md equivalent. The .md files can use YAML front-matter to be their own self-contained file.

**Comment by lachellel** [comment]
Thanks @konklone 

what editor do you use btw? I'm typically using atom (ubuntu)

**Comment by konklone** [comment]
I'm using Sublime Text 3. I'd like to give Atom a shot at some point, I'm told it's more fast and stable now!

**Pull request merged**